pkg/plugins/s3/source/iterator/snapshot_iterator.go

```diff
@@ -134,19 +128,18 @@ func (w *SnapshotIterator) Next(ctx context.Context) (record.Record, error) {
 	}
 
 	// create the record
-	output := record.Record{
+	output := sdk.Record{
 		Metadata: map[string]string{
 			"content-type": *object.ContentType,
 		},
-		Position: p.ToRecordPosition(),
-		Payload: record.RawData{
-			Raw: rawBody,
-		},
-		Key: record.RawData{
-			Raw: []byte(*key),
-		},
+		Position:  p.ToRecordPosition(),
+		Payload:   sdk.RawData(rawBody),
+		Key:       sdk.RawData(*key),
 		CreatedAt: *object.LastModified,
 	}
 
 	return output, nil
 }
+func (w *SnapshotIterator) Stop() {
+	// nothing to stop
+}
```

Comment:
Do we need this function?

Reply:
not really.. just added it so it would implement the Iterator interface